### PR TITLE
Make Contacts feature work and be more user friendly

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/gEAMmD0g/386-ability-to-embed-a-contact-via-markdown

This PR is to fix issues encountered by trying to use the Contacts feature added to Govspeak in 2016. Prior to this this feature was not used in any apps (evidenced by this feature not actually working properly) thus I don't consider any changes to this to be backwards compatibility breaks.

### CODE REVIEW

File: test/govspeak_contacts_test.rb [hunk 1]
```diff
@@ -74,17 +96,11 @@ def compress_html(html)
...
 
   test "contact with no postal address omits the address info" do
-    contact = build_contact(
-      recipient: nil,
-      street_address: nil,
-      locality: nil,
-      region: nil,
-      postal_code: nil,
-    )
+    contact = build_contact(post_addresses: [])
```
Comment (benthorner) [inline test/govspeak_contacts_test.rb]: Not really part of this PR, but I notice there's (effectively) a duplicate test about not finding a contact, and no test for the expected HTML when there's no email/phone/URL.

Comment (kevindew) [reply test/govspeak_contacts_test.rb]: https://github.com/alphagov/govspeak/blob/ec96ea5cfb334c814232f17f1bdc71b446f8254b/test/govspeak_contacts_test.rb#L86-L96

File: lib/govspeak/presenters/contact_presenter.rb [hunk 2]
```diff
@@ -1,24 +1,61 @@
...
+
+  private
+
+    def filter_post_addresses(addresses)
+      addresses.each do |address|
+        address.delete_if do |key, value|
+          # Not showing United Kingdom country is a "feature" lifted and shifted
+          # from Whitehall:
+          # https://github.com/alphagov/whitehall/blob/c67d53d80f9856549c2da1941a10dbb9170be494/lib/address_formatter/formatter.rb#L17
+          value = "" if key == "world_location" && value.strip == "United Kingdom"
```
[the remark this answers is not in the document]
Comment (kevindew) [reply lib/govspeak/presenters/contact_presenter.rb]: I've reworked this now. Hopefully it's easier now - didn't quite understand how your suggestion could work I'm afraid.

File: lib/govspeak.rb [hunk 3]
```diff
@@ -89,6 +98,15 @@ def extracted_links(website_root: nil)
       Govspeak::LinkExtractor.new(self, website_root: website_root).call
     end
 
+    def extract_contact_content_ids
```
[the remark this answers is not in the document]
Comment (kevindew) [reply lib/govspeak.rb]: As discussed offline - there doesn't seem to be a clear case for reusing this a lot and the arguments to do so are tricky (since they don't match the markdown syntax)

We've kept the UUID regex so that other things don't have to worry about these not being UUIDs - since they can be anything - but as discussed I've moved it it to a constant (which the linter made me freeze - I didn't even realise a regex was mutable in Ruby)

**Comment by kevindew** [comment]
Thanks @benthorner - sure if we have a mo tomorrow lets chat.
